### PR TITLE
Cancel sleep when context is cancelled

### PR DESCRIPTION
**Pull request by vmihailenco**
Fixes https://github.com/go-redis/redis/issues/1100
Closes https://github.com/go-redis/redis/issues/1107

### CODE REVIEW

File: internal/error.go [hunk 1]
```diff
@@ -9,10 +10,10 @@ import (
 )
 
 func IsRetryableError(err error, retryTimeout bool) bool {
-	if err == nil {
+	switch err {
+	case nil, context.Canceled, context.DeadlineExceeded:
```
Comment (kollektiv) [inline internal/error.go]: Is this needed? I would expect `context.Canceled|DeadlineExceeded` to return false in the previous code

File: cluster.go [hunk 2]
```diff
@@ -754,7 +754,9 @@ func (c *ClusterClient) process(ctx context.Context, cmd Cmder) error {
 		var err error
 
 		if attempt > 0 {
-			time.Sleep(c.retryBackoff(attempt))
+			if err := internal.Sleep(ctx, c.retryBackoff(attempt)); err != nil {
+				return err
```
Comment (kollektiv) [inline cluster.go]: Does there need to be a `cmd.setErr(err)` here and elsewhere?

File: internal/error.go (hunk 1 above)
Comment (vmihailenco) [reply internal/error.go]: It is not needed, but `err == context.Canceled` is fast(er) and it is also explicit.

File: cluster.go (hunk 2 above)
Comment (vmihailenco) [reply cluster.go]: It should be. It is just becomes hard to track when `cmd.setErr` should be called. I hope to address this in https://github.com/go-redis/redis/issues/1099

**Pull request merged**